src/renderer/wg_engine/tvgWgCompositor.cpp

```diff
@@ -258,7 +258,25 @@ void WgCompositor::blendImage(WgContext& context, WgRenderDataPicture* renderDat
 };
 
 
-// TODO: use direct mask applience
+void WgCompositor::blendScene(WgContext& context, WgRenderStorage* src, WgCompose* cmp)
+{
+    assert(currentTarget);
+    // cut viewport to screen dimensions
+    int32_t xmin = std::max(0, std::min((int32_t)width, cmp->aabb.x));
```

Comment:
maybe some function? exactly the same part of the code to create render region is used in the composeScene()